pmd-core/src/main/java/net/sourceforge/pmd/renderers/CodeClimateRenderer.java

```diff
@@ -54,7 +53,6 @@ private static String getPmdPropertiesURL() {
 
     @Override
     public void renderFileViolations(Iterator<RuleViolation> violations) throws IOException {
-        Writer writer = getWriter();
```

Comment:
That's a small trick to not trigger CloseResource rule: The field `writer` is anyway protected and available here, so we don't need to create a local variable. The rule looks on local variables. But maybe we should also check, that the resources are created in the method and not retrieved from somewhere else... so we probably should look for a constructor call and only then consider the local var